File: backend/app/models/notification.py
from datetime import datetime
from sqlalchemy import Column, Integer, String, Boolean, DateTime, Text
from app.core.database import Base


class Notification(Base):
    __tablename__ = "notifications"

    id = Column(Integer, primary_key=True, index=True)
    type = Column(String(32), nullable=False)
    title = Column(String(128), nullable=False)
    body = Column(Text, nullable=False, default="")
    is_read = Column(Boolean, default=False, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    # audience: 'admin' = admin only, 'all' = everyone, 'user' = specific user
    audience = Column(String(16), nullable=False, default="admin")
    target_user_id = Column(Integer, nullable=True)   # set when audience='user'

File: backend-go/models/notification.go
package models

import "time"

// Notification 系统通知记录。
// Audience 决定可见范围：admin / support / all / user（配合 TargetUserID）。
type Notification struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	Type         string    `gorm:"size:32;not null" json:"type"` // 通知类型，如 sms_failed
	Title        string    `gorm:"size:128;not null" json:"title"`
	Body         string    `gorm:"type:text;not null;default:''" json:"body"` // 通知正文
	IsRead       bool      `gorm:"not null;default:false" json:"is_read"`
	CreatedAt    time.Time `gorm:"not null" json:"created_at"`
	Audience     string    `gorm:"size:16;not null;default:admin" json:"audience"` // 受众：admin/support/all/user
	TargetUserID *uint     `json:"target_user_id,omitempty"`                       // audience=user 时指定目标用户
}

func (Notification) TableName() string { return "notifications" }
